backend/setup_secrets.bash

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+url='https://console.firebase.google.com/project/trogdors-29fa4/settings/serviceaccounts/databasesecrets'
+conf_dirs=(~/.xdg/config ~/.config)
+conf_name=hvz_secrets
+
+# Try to find an existing conf
+for dir in "${conf_dirs[@]}"; do
+  conf="${dir}/${conf_name}"
+  [[ -r "$conf" ]] && break
+done
+
+# Create a conf
+if ! [[ -r "$conf" ]]; then
+  for dir in "${conf_dirs[@]}"; do
+    [[ -w "$dir" ]] && break
+  done
+
+  if ! [[ -w "$dir" ]]; then
+    printf 'Was not able to locate a conf file nor a conf dir.\n'
+    exit 1
+  fi
+  conf="${dir}/${conf_name}"
+  printf 'Visit %s to get the Firebase secret.\n' "$url"
+  read -rp 'Firebase secret? ' fb_secret
+  read -rp 'Firebase email (not really used)? ' fb_email
+  printf 'For Ionic, you need the (1) token (2) app id (3) security tag.\n'
+  read -rp 'Ionic token? ' itok
+  read -rp 'Ionic app id? ' iapp
+  read -rp 'Ionic security tag? ' itag
+
+  {
+    printf '# This file is auto-generated by %s\n' "$0"
+    printf 'FIREBASE_SECRET = "%s"\n' "$fb_secret"
+    printf 'FIREBASE_EMAIL = "%s"\n' "$fb_email"
+    printf 'IONIC_TOKEN = "%s"\n' "$itok"
+    printf 'IONIC_APPID = "%s"\n' "$iapp"
+    printf 'IONIC_SECURITY_TAG  = "%s"\n' "$itag"
+  } > "$conf"
+
+fi
+
+if ! [[ -r "$conf" ]]; then
+  printf 'Could not read the conf file, %s.' "$conf"
+  exit 1
+fi
+
+dest="${0%/*}"
```

Comment:
This is POSIX paramter expansion and not uncommon. It's also faster and the recommended approach.